mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala

```diff
@@ -37,6 +37,31 @@ import org.apache.spark.sql.types._
  */
 private[feature] trait RFormulaBase extends HasFeaturesCol with HasLabelCol {
 
+  /**
+   * Param for how to order labels of string column. The first label after ordering is assigned
+   * an index of 0.
+   * Options are:
+   *   - 'frequencyDesc': descending order by label frequency (most frequent label assigned 0)
+   *   - 'frequencyAsc': ascending order by label frequency (least frequent label assigned 0)
+   *   - 'alphabetDesc': descending alphabetical order
+   *   - 'alphabetAsc': ascending alphabetical order
+   * Default is 'frequencyDesc'.
+   * When the ordering is set to 'alphabetDesc', `RFormula` drops the same category as R
+   * when encoding strings.
+   *
+   * @group param
+   */
+  @Since("2.3.0")
+  final val stringOrderType: Param[String] = new Param(this, "stringOrderType",
```

Comment:
What about ```orderTypeOfStringIndexer``` or any other suggestion? I think the param name ```stringOrderType``` in ```StringIndexer```   is clear enough, but ```RFormula``` involves lots of feature transformers across many steps, we should make users understand this param only takes effect on ```StringIndexer``` stage. And it's better to document more clear rather than copying corresponding section from ```StringIndexer```. cc @felixcheung